haskell/private/coverage_wrapper.sh.tpl

```diff
@@ -25,38 +27,40 @@ fi
 
 ERRORCOLOR='\033[1;31m'
 CLEARCOLOR='\033[0m'
-binary_path=$(rlocation {binary_path})
-hpc_path=$(rlocation {hpc_path})
-tix_file_path={tix_file_path}
-coverage_report_format={coverage_report_format}
-strict_coverage_analysis={strict_coverage_analysis}
-package_path={package_path}
+binary_path=$(rlocation "{binary_path}")
+hpc_path=$(rlocation "{hpc_path}")
+tix_file_path="{tix_file_path}"
+coverage_report_format="{coverage_report_format}"
+strict_coverage_analysis="{strict_coverage_analysis}"
+package_path="{package_path}"
 
 # either of the two expected coverage metrics should be set to -1 if they're meant to be unused
-expected_covered_expressions_percentage={expected_covered_expressions_percentage}
-expected_uncovered_expression_count={expected_uncovered_expression_count}
+expected_covered_expressions_percentage="{expected_covered_expressions_percentage}"
+expected_uncovered_expression_count="{expected_uncovered_expression_count}"
 
 # gather the hpc directories
 hpc_dir_args=""
-mix_file_paths={mix_file_paths}
+mix_file_paths="{mix_file_paths}"
 for m in "${mix_file_paths[@]}"
 do
   absolute_mix_file_path=$(rlocation $m)
-  hpc_parent_dir=$(dirname $absolute_mix_file_path)
-  trimmed_hpc_parent_dir=$(echo "${hpc_parent_dir%%.hpc*}")
+  hpc_parent_dir=$(dirname "$absolute_mix_file_path")
+  trimmed_hpc_parent_dir="${hpc_parent_dir%%.hpc*}"
   hpc_dir_args="$hpc_dir_args --hpcdir=$trimmed_hpc_parent_dir.hpc"
 done
 
 # gather the modules to exclude from the coverage analysis
 hpc_exclude_args=""
-modules_to_exclude={modules_to_exclude}
+modules_to_exclude="{modules_to_exclude}"
 for m in "${modules_to_exclude[@]}"
 do
   hpc_exclude_args="$hpc_exclude_args --exclude=$m"
 done
 
 # run the test binary, and then generate the report
 $binary_path "$@" > /dev/null 2>&1
+# Disabling SC2086: we want word splitting on hpc_dir_args and hpc_exclude_args
```

Comment:
Same, these should really be bash arrays

Reply:
Done in 46f554b41439c87de16cf6f0f17929dfe0251a5e.